### PR TITLE
fix incorrect handling of RestEndpointIOException in ListenerUtils

### PR DESCRIPTION
**Pull request by dmikhievich**

### CODE REVIEW

File: src/main/java/com/epam/reportportal/listeners/ListenersUtils.java [hunk 1]
```diff
@@ -49,7 +49,7 @@ private ListenersUtils() {
 	 * @param message
 	 */
 	public static void handleException(Exception exception, Logger logger, String message) {
-		if (ReportPortalException.class.isInstance(exception) || RestEndpointIOException.class.getClass().equals(exception.getClass())) {
+		if (ReportPortalException.class.isInstance(exception) || RestEndpointIOException.class.equals(exception.getClass())) {
```
Comment (avarabyeu) [inline src/main/java/com/epam/reportportal/listeners/ListenersUtils.java:52]: Agreed

File: src/test/java/com/epam/reportportal/listeners/ListenerUtilsTest.java [hunk 2]
```diff
@@ -34,4 +40,23 @@ public void getLaunchMode() {
 		Assert.assertEquals(DEFAULT, ListenersUtils.getLaunchMode("notvalid"));
 		Assert.assertEquals(DEBUG, ListenersUtils.getLaunchMode("Debug"));
 	}
+
+	@Test
+	public void handleException_restEndpointIOExceptionAndNullLogger() {
```
Comment (avarabyeu) [inline src/test/java/com/epam/reportportal/listeners/ListenerUtilsTest.java:46]: Do we really need this tests? What if logger is configured for files instead for system console?

**Review by avarabyeu — CHANGES_REQUESTED** [review]

File: src/test/java/com/epam/reportportal/listeners/ListenerUtilsTest.java (hunk 2 above)
Comment (dmikhievich) [reply src/test/java/com/epam/reportportal/listeners/ListenerUtilsTest.java:46]: Basically this test checks the case when the "logger" method argument is null just to be sure that the error messaging code is reachable:

```
if (logger != null) {
       logger.error(message, exception);
} else {
       System.out.println(exception.getMessage());
}
```

And I agree that it makes sense to cover the code branch with output to logger too.

Comment (dmikhievich) [reply src/test/java/com/epam/reportportal/listeners/ListenerUtilsTest.java:46]: Additional test was added.

**Review by avarabyeu — APPROVED** [review]

**Pull request merged**